src/qt/guiutil_font.cpp

```diff
@@ -0,0 +1,596 @@
+// Copyright (c) 2014-2025 The Dash Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <qt/guiutil_font.h>
+
+#include <tinyformat.h>
+#include <util/system.h>
+
+#include <qt/guiutil.h>
+
+#include <QApplication>
+#include <QDebug>
+#include <QFontDatabase>
+#include <QFontMetrics>
+#include <QPointer>
+#include <QWidget>
+
+#include <cmath>
+#include <map>
+#include <memory>
+#include <utility>
+
+namespace GUIUtil {
+/** loadFonts stores the SystemDefault font in osDefaultFont to be able to reference it later again */
+static std::unique_ptr<QFont> osDefaultFont;
+/** Font related default values. */
+const FontSettings g_font_defaults{};
+
+/** Font related variables. */
+struct FontAttributes
+{
+    FontFamily family;
+    FontWeight weight;
+    bool is_italic;
+    int size;
+};
+
+// Application font family. May be overwritten by -font-family.
+static FontFamily fontFamily = g_font_defaults.family;
+// Application font scale value. May be overwritten by -font-scale.
+static int fontScale = g_font_defaults.scale;
+// Contains the weight settings separated for all available fonts
+static std::map<FontFamily, std::pair<QFont::Weight, QFont::Weight>> mapDefaultWeights;
+static std::map<FontFamily, std::pair<QFont::Weight, QFont::Weight>> mapWeights;
+// Contains all widgets and its font attributes (family, weight, italic, size) with font changes due to GUIUtil::setFont
+static std::map<QPointer<QWidget>, FontAttributes> mapFontUpdates;
+// Contains a list of supported font weights for all members of GUIUtil::FontFamily
+static std::map<FontFamily, std::vector<QFont::Weight>> mapSupportedWeights;
+
+//! Map between font weights, Montserrat's convention and italic availability
+static const std::map<QFont::Weight, std::pair<QString, /*can_italic=*/bool>> mapMontserrat{{
+    {QFont::Black, {"Black", true}},
+    {QFont::Bold, {"Bold", true}},
+    {QFont::DemiBold, {"SemiBold", true}},
+    {QFont::ExtraBold, {"ExtraBold", true}},
+    {QFont::ExtraLight, {"ExtraLight", true}},
+    {QFont::Light, {"Light", true}},
+    {QFont::Medium, {"Medium", true}},
+    {QFont::Normal, {"Regular", false}},
+    {QFont::Thin, {"Thin", true}},
+}};
+
+FontFamily fontFamilyFromString(const QString& strFamily)
+{
+    for (const auto& [family, family_name, _] : AVAILABLE_FONTS) {
+        if (strFamily == family_name) {
+            return family;
+        }
+    }
+    throw std::invalid_argument(strprintf("Invalid font-family: %s", strFamily.toStdString()));
+}
+
+QString fontFamilyToString(FontFamily family)
+{
+    assert(family <= FontFamily::MaxVal);
+    return AVAILABLE_FONTS[static_cast<size_t>(family)].family_name;
```

Comment:
nit: use here `ToUnderlying`

```suggestion
    return AVAILABLE_FONTS[ToUnderlying(family)].family_name;
```

and also
```diff
+#include <util/underlying.h>
```